src/Arc4u.Standard.OAuth2.Client/Security/Principal/AppPrincipalFactory.cs

```diff
@@ -28,84 +27,72 @@ public class AppPrincipalFactory : IAppPrincipalFactory
+        private readonly IApplicationContext _applicationContext;
+        private readonly ILogger<AppPrincipalFactory> _logger;
 
-        private bool copyClaimsFromCache = false;
-        private List<ClaimDto> cachedClaims;
-
-        public AppPrincipalFactory(INetworkInformation networkInformation, ISecureCache claimsCache, ICacheKeyGenerator cacheKeyGenerator, IContainerResolve container, IApplicationContext applicationContext, ILogger<AppPrincipalFactory> logger)
+        public AppPrincipalFactory(IContainerResolve container, INetworkInformation networkInformation, ISecureCache claimsCache, ICacheKeyGenerator cacheKeyGenerator, IApplicationContext applicationContext, ILogger<AppPrincipalFactory> logger)
         {
-            NetworkInformation = networkInformation;
-            ClaimsCache = claimsCache;
-            CacheKeyGenerator = cacheKeyGenerator;
-            Container = container;
-            ApplicationContext = applicationContext;
+            _container = container;
+            _networkInformation = networkInformation;
+            _claimsCache = claimsCache;
+            _cacheKeyGenerator = cacheKeyGenerator;
+            _applicationContext = applicationContext;
             _logger = logger;
         }
 
-        private INetworkInformation NetworkInformation { get; set; }
-
-        private ICache ClaimsCache { get; set; }
-
-        private ClaimsIdentity Identity { get; set; }
-
-        private IContainerResolve Container { get; set; }
-
-        private ICacheKeyGenerator CacheKeyGenerator { get; set; }
-
-        private IApplicationContext ApplicationContext { get; set; }
-
-        private readonly ILogger<AppPrincipalFactory> _logger;
-
         public async Task<AppPrincipal> CreatePrincipal(Messages messages, object parameter = null)
         {
-            Identity = new ClaimsIdentity("OAuth2Bearer", System.Security.Claims.ClaimTypes.Upn, ClaimsIdentity.DefaultRoleClaimType);
-
             return await CreatePrincipal(DefaultSettingsResolveName, messages, parameter).ConfigureAwait(true);
         }
 
         public async Task<AppPrincipal> CreatePrincipal(string settingsResolveName, Messages messages, object parameter = null)
         {
-            var settings = Container.Resolve<IKeyValueSettings>(settingsResolveName);
-
+            var settings = _container.Resolve<IKeyValueSettings>(settingsResolveName);
             return await CreatePrincipal(settings, messages, parameter);
         }
 
         public async Task<AppPrincipal> CreatePrincipal(IKeyValueSettings settings, Messages messages, object parameter = null)
         {
-            Identity = new ClaimsIdentity("OAuth2Bearer", System.Security.Claims.ClaimTypes.Upn, ClaimsIdentity.DefaultRoleClaimType);
+            var identity = new ClaimsIdentity("OAuth2Bearer", System.Security.Claims.ClaimTypes.Upn, ClaimsIdentity.DefaultRoleClaimType);
 
             if (null == settings) throw new ArgumentNullException(nameof(settings));
```

Comment:
I also propose to use ArgumentNullException.ThrowIfNull

Reply:
The assembly is in netstandard 2.0 / 2.1
The method is available in .net 6 / .net 7